### PR TITLE
Kanban backlogs feature

### PR DESCRIPTION
**Pull request by Tinsae**
This is stable version for now

### CODE REVIEW

File: ProjectForce/force-app/main/default/aura/KanbanCard/KanbanCardController.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+({
+	doInit: function(component, event, helper){
+		//component.set("v.labelColor", component.get("v.backlog.CardLabel__r.Color__c"));
+        //var backlog = component.get("v.backlog");
+        // action.setParams({backlog: backlog });
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/KanbanCard/KanbanCardController.js]: Please take out commented code.

File: ProjectForce/force-app/main/default/aura/KanbanCardLabel/KanbanCardLabelController.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+({
+    doInit: function (component, event, helper) {
+        var labelIndex = component.get("v.labelIndex");
+        var color = "gray";
+        if (labelIndex == 1) {
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/KanbanCardLabel/KanbanCardLabelController.js:5]: What does 1,2,3 signify?

File: ProjectForce/force-app/main/default/aura/KanbanProject/KanbanProject.cmp [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<aura:component controller="KanbanProjectApexClass"
+    implements="force:appHostable,flexipage:availableForAllPageTypes,flexipage:availableForRecordHome,force:hasRecordId,force:lightningQuickAction"
+    access="global">
+    <!-- resources -->
+    <ltng:require styles="{!$Resource.bulma + '/bulma-0.4.2/css/bulma.css'}" />
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/KanbanProject/KanbanProject.cmp]: What's this?

File: ProjectForce/force-app/main/default/aura/KanbanProject/KanbanProject.cmp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <aura:attribute name="backlogs" type="Backlog__c[]" />
+    <!-- project id stored as a string -->
+    <aura:attribute name="project" type="String" />
+
+    <!-- event handlers -->
+    <aura:handler name="init" action="{!c.doInit}" value="{!this}"></aura:handler>
+    <aura:handler name="droppedOnColumn" event="c:droppedOnColumn" action="{!c.droppedHandler}"></aura:handler>
+
+    <div class="columns">
+        <aura:if isTrue="{!v.backlogs.length}">
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/KanbanProject/KanbanProject.cmp:12]: Does this mean that we are currently limited to three columns?

File: ProjectForce/force-app/main/default/aura/KanbanProject/KanbanProjectController.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        // Send action off to be executed
+        $A.enqueueAction(action);
+    },
+
+    droppedHandler: function (component, event, helper) {
+        var title = event.getParam('title');
+        var backlog = event.getParam('backlog');
+        var listOfBacklogs = component.get('v.backlogs');
+
+        var theBacklog = listOfBacklogs.find(function (el) {
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/aura/KanbanProject/KanbanProjectController.js:45]: I'd add a comment here to explain what this is doing.

File: ProjectForce/force-app/main/default/classes/KanbanProjectApexClass.cls [hunk 6]
```diff
@@ -0,0 +1,21 @@
+public with sharing class KanbanProjectApexClass {
+    @AuraEnabled
+    public static List<Backlog__c> getBacklogs(String project) {
+        return [
+            SELECT 
+                Id, Name, Story__c, NumberOfCardLabels__c,
```
Comment (mruizrevature) [inline ProjectForce/force-app/main/default/classes/KanbanProjectApexClass.cls:6]: Are backlogs currently limited to three labels?

**Review by mruizrevature — CHANGES_REQUESTED** [review]
Looks good. Just take a look at the requested changes.

File: ProjectForce/force-app/main/default/classes/KanbanProjectApexClass.cls (hunk 6 above)
Comment (Tinsae) [reply ProjectForce/force-app/main/default/classes/KanbanProjectApexClass.cls:6]: yes.

**Comment by Tinsae** [comment]
Kanban Component requires a project in order to render. So provide "project" when loading this component from other components.
Currently there is a tab "MangeProjectKanban" which loads backlogs without any projects

**Review by mruizrevature — APPROVED** [review]

**Comment by mruizrevature** [comment]
Nice job

**Pull request merged**